core/shared/src/test/scala/fs2/concurrent/ChannelSuite.scala

```diff
@@ -278,9 +278,17 @@ class ChannelSuite extends Fs2Suite {
   }
 
   test("sendAndClose closes right after sending the last element in synchronous case") {
-    val channel = Channel.synchronous[IO, Int]
+    val test = Channel
+      .synchronous[IO, Int]
+      .flatMap { ch =>
+        ch.stream
+          .concurrently(Stream.emit(0).covary[IO].evalMap(ch.sendAndClose))
+          .compile
+          .drain *> ch.isClosed
+      }
+      .assertEquals(true)
```

Comment:
Nice! Here's a little refactoring of the same idea: draining the channel concurrently in the background.

```suggestion
    val test = Channel.synchronous[IO, Int].flatMap { ch =>
      ch.stream.compile.drain.background.surround {
        ch.sendAndClose(0) *> ch.isClosed.assert
      }
```